### ISSUE TITLE
dynamic inventory does not populate name of attached disks

### ISSUE THREAD

**Issue by DII-dsward** [ansible/ansible_collections_azure#678]
Need to tag our managed disks for cost analysis, but the dynamic inventory does not return the name of the managed disks so that we can iterate them as vars. It does return some information, but not the name var.

Example of what dynamic inventory is returning related to data disks.
```yaml
data_disks:
- {caching: ReadOnly, disk_size_gb: 128, lun: 0, managed_disk_type: StandardSSD_LRS}
- {caching: ReadOnly, disk_size_gb: 1024, lun: 1, managed_disk_type: Standard_LRS}
- {caching: ReadOnly, disk_size_gb: 1024, lun: 2, managed_disk_type: Standard_LRS}
- {caching: ReadOnly, disk_size_gb: 1024, lun: 3, managed_disk_type: Standard_LRS}
- {caching: ReadOnly, disk_size_gb: 512, lun: 4, managed_disk_type: Standard_LRS}
- {caching: None, disk_size_gb: 512, lun: 5, managed_disk_type: Standard_LRS}
- {caching: None, disk_size_gb: 512, lun: 6, managed_disk_type: Standard_LRS}
```

##### SUMMARY
the dynamic inventory yaml
```yaml
---
# https://github.com/ansible-collections/azure/blob/dev/plugins/inventory/azure_rm.py
plugin: azure.azcollection.azure_rm
location:
  - southcentralus
```

##### ISSUE TYPE
- Bug Report

##### COMPONENT NAME
- plugin: azure.azcollection.azure_rm

##### EXPECTED OUTPUT
```yaml
data_disks:
- {name: vmname-osdisk, caching: ReadOnly, disk_size_gb: 128, lun: 0, managed_disk_type: StandardSSD_LRS}
- {name: vmname-disk1, caching: ReadOnly, disk_size_gb: 1024, lun: 1, managed_disk_type: Standard_LRS}
- {name: vmname-disk2, aching: ReadOnly, disk_size_gb: 1024, lun: 2, managed_disk_type: Standard_LRS}
- {name: vmname-cache, caching: ReadOnly, disk_size_gb: 1024, lun: 3, managed_disk_type: Standard_LRS}
- {name: vmname-data1, caching: ReadOnly, disk_size_gb: 512, lun: 4, managed_disk_type: Standard_LRS}
- {name: vmname-sql1, caching: None, disk_size_gb: 512, lun: 5, managed_disk_type: Standard_LRS}
- {name: vmname-1, caching: None, disk_size_gb: 512, lun: 6, managed_disk_type: Standard_LRS}
```

**Comment by Fred-sun** [comment]
@DII-dsward Thank you for reporting this problem, we will check and fix it as soon as possible. Thank you very much!

**Comment by Fred-sun** [comment]
fixes by #687
